masonry/src/widgets/canvas.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright 2025 the Xilem Authors and the Druid Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! A canvas widget.
+
+use accesskit::{Node, Role};
+use masonry_core::core::{ArcStr, ChildrenIds};
+use tracing::{Span, trace_span};
+use vello::Scene;
+use vello::kurbo::Size;
+
+use crate::core::{
+    AccessCtx, BoxConstraints, LayoutCtx, PaintCtx, PropertiesMut, PropertiesRef, RegisterCtx,
+    Widget, WidgetId, WidgetMut,
+};
+
+/// A widget allowing custom drawing.
+///
+/// A canvas takes a painter callback; every time the canvas is repainted, that callback
+/// in run with a [`Scene`].
+/// That Scene is then used as the canvas' contents.
+#[derive(Default)]
+pub struct Canvas {
+    alt_text: ArcStr,
```

Comment:
As mentioned in #1519, "empty alt text" and "no alt text" are different semantic values, so we should make this an option.

Reply:
Should be solved